server/graphql/general/src/lib.rs

```diff
@@ -148,67 +153,92 @@ impl GeneralQueries {
     ) -> Result<LogResponse> {
         logs(ctx, page, filter, sort)
     }
-}
 
-#[derive(Default, Clone)]
-pub struct ServerAdminQueries;
+    /// Available without authorisation in operational and initialisation states
+    pub async fn sync_state(&self, ctx: &Context<'_>) -> Result<SyncStateType> {
+        sync_state(ctx)
+    }
 
-#[Object]
-impl ServerAdminQueries {
-    /// Retrieves a new auth bearer and refresh token
-    /// The refresh token is returned as a cookie
-    pub async fn server_settings(&self, ctx: &Context<'_>) -> Result<ServerSettingsResponse> {
-        get_server_settings(ctx, false)
+    pub async fn latest_sync_status(
+        &self,
+        ctx: &Context<'_>,
+    ) -> Result<Option<FullSyncStatusNode>> {
+        latest_sync_status(ctx, true)
+    }
+
+    pub async fn number_of_records_in_push_queue(&self, ctx: &Context<'_>) -> Result<u64> {
+        number_of_records_in_push_queue(ctx)
     }
 
-    /// Restarts the server
-    pub async fn server_restart(&self, ctx: &Context<'_>) -> Result<RestartNode> {
-        server_restart(ctx, false).await
+    pub async fn sync_settings(&self, ctx: &Context<'_>) -> Result<SyncSettingsResponse> {
+        sync_settings(ctx, true)
     }
 }
+
 #[derive(Default, Clone)]
-pub struct ServerAdminMutations;
+pub struct GeneralMutations;
```

Comment:
The partitioning we used for graphql crate was only really for compilation time, splitting graphql into seperate crates allows for incremental builds and faster compilation or more responsive rust analyzer. Wanted to keep to that partitioning strategy, rather then introducing another layer of partitioning (which was there and I reverted it)

Reply:
So basically since it's coming from `general` crate it's just `GeneralMutation`, happy to reconsider but would rather a more simple rule to use for naming/grouping graphql (group all mutations/queries based on he crate)